runner/runner print formatting for process status

dgassaway opened the pull request:
Added String() methods to struct types: runner.ProcessStatus and runner.Command so these print human-readable.

Please note that this removed all uses of the vendor lib github.com/luci/go-render which has been removed - looking for feedback on this.

dbentley commented on runner/runner.go:
```diff
@@ -34,7 +34,7 @@ const (
 
 func (p ProcessState) IsDone() bool {
-	return p == COMPLETE || p == FAILED || p == ABORTED || p == TIMEDOUT
+	return p == COMPLETE || p == FAILED || p == ABORTED || p == TIMEDOUT || p == BADREQUEST
```
As discussed in person:

1) BADREQUEST should probably not be IsDone, because a retry could work
2) BADREQUEST is weird, and maybe we should just return an error instead of a ProcessStatus...

jasonbs10 replied:
I'm kind of regretting BADREQUEST. I initially thought it would be a good idea in worker.thrift but didn't really want it in runner.go. We should consider converting it to an error as Dan said.

dbentley commented on runner/runner.go:
```diff
@@ -78,6 +78,24 @@ type Command struct {
+	sargs = append(sargs, c.Argv, c.Timeout)
+
+	if len(c.EnvVars) > 0 {
+		sfmt += "\tEnv:\n"
```
Adding to sfmt is weird... maybe just do multiple Sprintfs so you don't need to store the fmt string at all?

jasonbs10 commented on .gitmodules:
```diff
@@ -34,10 +34,6 @@
 	path = vendor/golang.org/x/sys
 	url = https://go.googlesource.com/sys
 	branch = 62bee037599929a6e9146f29d10dd5208c43507d
-[submodule "vendor/github.com/luci/go-render"]
```
I think we want to keep this. What was the rationale for removing this entirely?

jasonbs10 commented on runner/runner_test.go:
```diff
@@ -0,0 +1,72 @@
+
+func TestCommandStringSimple(t *testing.T) {
+	c := runner.Command{
+		Argv:       []string{"cat", "cats.in", "|", "cat.bin", ">", "cats.out"},
```
If tests double as documentation, I don't think we want to pipe/redirect here. AFAIK that's not supported in golang unless it's a single arg passed to a bash binary.

dbentley replied:
Good catch Jason; I agree.

dbentley commented on runner/runner_test.go:
```diff
@@ -0,0 +1,72 @@
+package runner_test
```
Maybe move this to status_test.go?

Also, does it need to be in a separate package (runner_test instead of runner)?

dgassaway commented:
* Removed submodule removal for potential future reuse (was not sure if we would scrub unused dependencies).
* Refactored String() method string building style
* Moved runner_test to status_test per feedback, although I'm not sure about this one - this is directly testing code implemented in runner.go.  I went with runner_test as the package name vs. runner per cloud_test code and some community recs on black-box test pkg names

dgassaway commented:
* Updated command print test to be a valid command

dbentley commented:
Hrm, I think this is testing status, but maybe we should just move the status definition into status.go and make runner.go have the definition of just the runner?

I used to prefer the _test package style name, but found that they're kinda rarely used. As a team (but before you joined), we talked about it and said it felt like more overhead and we would prefer to not use them when not required. If you feel strongly, we can have this discussion again at our tech discussion meeting.

Also, I realize now looking at the code that maybe using a bytes.Buffer would be even simpler for what you're trying to do, of just keep writing things to an output. Cf. https://godoc.org/bytes#example-Buffer
